.clang-tidy

```diff
@@ -1,2 +1,124 @@
-Checks:          'clang-diagnostic-*,clang-analyzer-*,-*,cert-*,cppcoreguidelines-*,clang-analyzer-*'
+---
+Checks:
+  - -*
+  - cert-*
+  - cppcoreguidelines-*
+  - clang-analyzer-*'
 WarningsAsErrors: ''
```

Comment:
> this PR is an opportunity to decide on an initially minimal-and-constrainted set of checks which will fail the `clang-tidy` task on error.

To check: is this PR asking reviewers to propose this set of checks? That may take time for me.